src/Build/Evaluation/Evaluator.cs

```diff
@@ -604,7 +604,10 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
         /// </summary>
         private void Evaluate(ILoggingService loggingService, BuildEventContext buildEventContext)
         {
-            string projectFile;
+            string projectFile = String.IsNullOrEmpty(_projectRootElement.ProjectFileLocation.File) ? "(null)" : _projectRootElement.ProjectFileLocation.File;
+
+            _evaluationLoggingContext = new EvaluationLoggingContext(loggingService, buildEventContext, projectFile);
```

Comment:
Some tests are failing because they're assuming a certain order (e.g. that EvaluationStarted would be the first log entry). I've had to move the creation of `_evaluationLoggingContext` to the beginning of the `Evaluate` method. Is it okay to move the following lines to this location as well?

```csharp
#if (!STANDALONEBUILD)
            CodeMarkers.Instance.CodeMarker(CodeMarkerEvent.perfMSBuildProjectEvaluatePass0End);
#endif

                _evaluationLoggingContext.LogProjectEvaluationStarted();
```